---
 sbin/bsdlabel/bsdlabel.c |   13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

--- a/sbin/bsdlabel/bsdlabel.c
+++ b/sbin/bsdlabel/bsdlabel.c
@@ -60,7 +60,9 @@ __FBSDID("$FreeBSD$");
 #include <sys/file.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
+#if HAVE_BSD_DISKLABEL
 #include <sys/disk.h>
+#endif
 #define DKTYPENAMES
 #define FSTYPENAMES
 #define MAXPARTITIONS	26
@@ -69,7 +71,7 @@ __FBSDID("$FreeBSD$");
 #include <unistd.h>
 #include <string.h>
 #include <stdio.h>
-#include <libgeom.h>
+//#include <libgeom.h>
 #include <stdlib.h>
 #include <signal.h>
 #include <stdarg.h>
@@ -720,10 +722,12 @@ editit(void)
 
 	omask = sigblock(sigmask(SIGINT)|sigmask(SIGQUIT)|sigmask(SIGHUP));
 	while ((pid = fork()) < 0) {
+#if HAVE_BSD_ERRNO
 		if (errno == EPROCLIM) {
 			warnx("you have too many processes");
 			return(0);
 		}
+#endif
 		if (errno != EAGAIN) {
 			warn("fork");
 			return(0);
@@ -1508,13 +1512,18 @@ getvirginlabel(void)
 	 * to get any good ideas from the device, construct something
 	 * which is IBM-PC friendly.
 	 */
+#if HAVE_BSD_DISKLABEL
 	if (ioctl(f, DIOCGFWSECTORS, &u) == 0)
 		loclab.d_nsectors = u;
 	else
+#endif
 		loclab.d_nsectors = 63;
+#if HAVE_BSD_DISKLABEL
 	if (ioctl(f, DIOCGFWHEADS, &u) == 0)
 		loclab.d_ntracks = u;
-	else if (loclab.d_secperunit <= 63*1*1024)
+	else
+#endif
+	if (loclab.d_secperunit <= 63*1*1024)
 		loclab.d_ntracks = 1;
 	else if (loclab.d_secperunit <= 63*16*1024)
 		loclab.d_ntracks = 16;
